tests/Unit/PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Test_Sinusoid.cpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "PointwiseFunctions/AnalyticData/ScalarAdvection/Sinusoid.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/ScalarAdvection/Sinusoid.hpp"
+
+namespace {
+void test_sinusoid_solution(const gsl::not_null<std::mt19937*> generator,
```

Comment:
I'd recommend looking at something like `Test_TovStar.cpp` for an idea of how we do tests nowadays. The Burgers code is very dated. The things that seem to be missing are serialization tests (and even TOV doesn't do that super well, it's better to feed the deserialized object through the full tests), copy and move semantics, equivalence operators, and creating from options. `Test_ShuOsherTube.cpp` might be the most modern example I can think of off the top of my head.

Reply:
Okay! I'll take a look at those.